### PR TITLE
bugfix: copy the map while SetLabelSelector

### PR DESCRIPTION
**Pull request by xzxiong**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #https://github.com/matrixorigin/MO-Cloud/issues/2876

## What this PR does / why we need it:
changes:
- copy the src map while SetLabelSelector. Make sure NO change the input variable.

### CODE REVIEW

**Comment by matrix-meow** [comment]
@xzxiong Thanks for your contributions!

### Pull Request Review:

#### Title:
The title of the pull request is clear and concise, indicating that it is a bug fix related to copying a map while setting a label selector.

#### Body:
The body of the pull request provides relevant information about the type of PR (BUG), the specific issue being addressed (#2876), and the purpose of the changes made, which is to copy the source map while setting the label selector to ensure no changes to the input variable.

#### Changes Made:
1. In the `SetLabelSelector` function in `db_holder.go`:
   - Previously, the function was directly assigning the input `labels` map to a global variable `gLabels`, which could lead to unintended modifications of the original map.
   - The change now creates a new map `gLabels` with an additional capacity to store the "account" key-value pair and then copies all key-value pairs from the input `labels` map to `gLabels`.
   - This ensures that the original `labels` map remains unchanged.

2. In `db_holder_test.go`:
   - A new test `TestSetLabelSelector` has been added to verify the behavior of the `SetLabelSelector` function.
   - The test checks if the function correctly sets the labels and if the global label selector `gLabels` is updated accordingly.

#### Feedback and Suggestions:
1. **Deep Copy vs. Shallow Copy:**
   - While the current implementation creates a new map and copies the key-value pairs, it performs a shallow copy. If the values in the map are reference types (e.g., slices, maps), modifying them in the original map will still affect the copied map.
   - To ensure a deep copy and prevent unintended modifications, consider recursively copying nested structures if necessary.

2. **Error Handling in Tests:**
   - The test cases should include error handling mechanisms to handle any failures during the test execution.
   - Add assertions to check for errors returned by functions like `SetLabelSelector` to ensure proper test coverage.

3. **Consistency in Test Naming:**
   - Ensure consistency in test naming conventions. For example, the test `TestSetLabelSelector` could follow a naming pattern like `TestSetLabelSelector_NormalCase` to clearly indicate the scenario being tested.

4. **Use Testify Assertions:**
   - The test currently uses `t.Errorf` for assertions. Consider using assertions provided by the `require` package from `github.com/stretchr/testify/require` for clearer failure messages and better test readability.

5. **Optimization:**
   - Instead of manually copying key-value pairs in the `SetLabelSelector` function, you can use a library like `github.com/ulule/deepcopier` to perform deep copying of maps efficiently.
   - This can simplify the code and reduce the risk of errors in the copying process.

6. **Documentation:**
   - Ensure that the code changes are well-documented, especially the rationale behind choosing a deep or shallow copy approach and any potential implications.

By addressing these suggestions, the codebase can be improved in terms of correctness, maintainability, and efficiency.

**Review by aylei — APPROVED** [review]
lgtm

File: pkg/util/export/etl/db/db_holder.go [hunk 1]
```diff
@@ -307,8 +307,11 @@ func SetLabelSelector(labels map[string]string) {
 	if len(labels) == 0 {
 		return
 	}
-	labels["account"] = "sys"
-	gLabels = labels
+	gLabels = make(map[string]string, len(labels)+1)
```
Comment (aylei) [inline pkg/util/export/etl/db/db_holder.go:310]: Will we mutate the gLabels returned by `GetLabelSelector`?

Comment (aylei) [reply pkg/util/export/etl/db/db_holder.go:310]: discussed offline

**Review by sukki37 — APPROVED** [review]

**Pull request merged**